### PR TITLE
Create and import flow

### PR DESCRIPTION
**Pull request by Kamill90**
## What does this PR do?
Allows subscribing just created or imported wallet through a new screen.
*(A few words about the PR)*

#### *Any background context you want to provide?*
Allows to add email in settings screen when no wallets available (or with skip choosing wallets)

## Todos:

- [ ] Checked on iOS
- [x] Checked on Android

### CODE REVIEW

File: src/screens/DeleteEntity/EntityScreen.tsx [hunk 1]
```diff
@@ -42,23 +46,26 @@ export const DeleteEntityScreen = ({
...
       }
       header={<Header isBackArrow title={title} />}
     >
       <Text style={styles.title}>{subtitle}</Text>
-      <Text style={styles.description}>
-        {i18n.wallets.deleteWallet.description1} {name}
-        {i18n.wallets.deleteWallet.description2}
-      </Text>
+      {typeof description === 'string' ? <Text style={styles.description}>{description}</Text> : description}
+      {!!note && (
```
Comment (KedziaPawel) [inline src/screens/DeleteEntity/EntityScreen.tsx:58]: this is a generic component so it should take as screen specific values. It should take props as Footer component and the whole component should be passed

File: src/screens/DeleteEntity/EntityScreen.tsx [hunk 2]
```diff
@@ -42,23 +46,26 @@ export const DeleteEntityScreen = ({
...
         </View>
       }
       header={<Header isBackArrow title={title} />}
     >
       <Text style={styles.title}>{subtitle}</Text>
-      <Text style={styles.description}>
-        {i18n.wallets.deleteWallet.description1} {name}
-        {i18n.wallets.deleteWallet.description2}
-      </Text>
+      {typeof description === 'string' ? <Text style={styles.description}>{description}</Text> : description}
```
Comment (KedziaPawel) [inline src/screens/DeleteEntity/EntityScreen.tsx:57]: The same goes for the description, the component should accept the whole component as the description and don't care what type of data is rendering

File: src/screens/CreateWalletScreen.tsx [hunk 3]
```diff
@@ -88,21 +87,41 @@ export class CreateWalletScreen extends React.PureComponent<Props, State> {
...
           secret: w.getSecret(),
-          onButtonPress: isNotificationEmailSet
+          onButtonPress: !!email
             ? () =>
-                this.props.navigation.navigate(Route.ReceiveNotificationsConfirmation, {
-                  address: email, // TODO
-                  flowType: ConfirmAddressFlowType.RECEIVE_NOTIFICATIONS_CONFIRMATION_IMPORT,
+                navigation.navigate(Route.Entity, {
+                  title: i18n.notifications.notifications,
+                  subtitle: i18n.notifications.getNotification,
```
Comment (KedziaPawel) [inline src/screens/CreateWalletScreen.tsx:109]: Zrzut ekranu 2021-01-19 o 15 14 57
i don't see the email named passed and the bold text

File: src/screens/ImportWalletScreen.tsx [hunk 4]
```diff
@@ -156,13 +163,29 @@ export class ImportWalletScreen extends PureComponent<Props, State> {
...
-          //     })
-          //   : // TODO in import wallet logic
-          this.showSuccessImportMessageScreen();
+          this.props.navigation.navigate(Route.CreateWalletSuccess, {
+            secret: newWallet.getSecret(),
+            onButtonPress: !!email
+              ? () =>
+                  this.props.navigation.navigate(Route.Entity, {
+                    title: i18n.notifications.notifications,
+                    subtitle: i18n.notifications.getNotification,
```
Comment (KedziaPawel) [inline src/screens/ImportWalletScreen.tsx:172]: same as in create

File: src/screens/ImportWalletScreen.tsx [hunk 5]
```diff
@@ -21,7 +21,7 @@ import { CreateMessage, MessageType } from 'app/helpers/MessageCreator';
 import { withCheckNetworkConnection, CheckNetworkConnectionCallback } from 'app/hocs';
```
[the remark this answers is not in the document]
Comment (Kamill90) [reply src/screens/ImportWalletScreen.tsx:21]: do you have a solution suggestion? Once I request checking if newWallet is subscribed, the store is updated after 'importWallet' is executed. I don't want to implement it in the components lifecycle, do you have an idea how to handle it with saga?

[the remark this answers is not in the document]
Comment (KedziaPawel) [reply src/screens/ImportWalletScreen.tsx:21]: yes, we shouldn't call api in the component
maybe add:

importWallet(newWallet, {
        onSuccess: (wallet) => {
           checkSubscription([wallet], onSuccess: (ids) => {
           //check if ids includes walle.id, if no, navigate to notifications
           })
        }

you would have to add to checkSubscriptionSaga meta.onSuccess(ids)

File: ios/GoldWallet.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -401,10 +401,6 @@
 							};
 						};
 					};
-					A281C085253846110012082E = {
```
Comment (KedziaPawel) [inline ios/GoldWallet.xcodeproj/project.pbxproj:404]: i think we should keep automating provisioning

File: src/screens/ImportWalletScreen.tsx [hunk 7]
```diff
@@ -154,15 +171,30 @@ export class ImportWalletScreen extends PureComponent<Props, State> {
...
       importWallet(newWallet, {
         onSuccess: () => {
-          // isNotificationEmailSet
-          //   ? this.props.navigation.navigate(Route.ReceiveNotificationsConfirmation, {
-          //       address: email,
-          //       flowType: ConfirmAddressFlowType.RECEIVE_NOTIFICATIONS_CONFIRMATION_IMPORT,
-          //     })
-          //   : // TODO in import wallet logic
-          this.showSuccessImportMessageScreen();
+          this.props.navigation.navigate(Route.CreateWalletSuccess, {
```
Comment (KedziaPawel) [inline src/screens/ImportWalletScreen.tsx:177]: check the flow in wimsical i dont see CreateWalletSuccess there

File: src/screens/ImportWalletScreen.tsx [hunk 8]
```diff
@@ -154,15 +171,30 @@ export class ImportWalletScreen extends PureComponent<Props, State> {
       });
     } else {
       newWallet.setLabel(this.state.label || i18n.wallets.import.imported + ' ' + newWallet.typeReadable);
+      checkSubscription([newWallet], email);
```
Comment (KedziaPawel) [inline src/screens/ImportWalletScreen.tsx:174]: so how it supposed to work?  `checkSubscription`  is executed and below the user is always redirected to notifications if has an email, no matter if the wallet already is already subscribed

File: src/screens/ImportWalletScreen.tsx [hunk 9]
```diff
@@ -521,11 +553,13 @@ export class ImportWalletScreen extends PureComponent<Props, State> {
 const mapStateToProps = (state: ApplicationState) => ({
   wallets: walletsSelectors.wallets(state),
   isNotificationEmailSet: isNotificationEmailSet(state),
-  email: email(state),
+  email: storedEmail(state),
+  subscribedIds: subscribedIds(state),
```
Comment (KedziaPawel) [inline src/screens/ImportWalletScreen.tsx:557]: unused

File: src/screens/ImportWalletScreen.tsx (hunk 8 above)
Comment (Kamill90) [reply src/screens/ImportWalletScreen.tsx:174]: that's what I was asking you, thanks

File: src/screens/ImportWalletScreen.tsx (hunk 7 above)
Comment (Kamill90) [reply src/screens/ImportWalletScreen.tsx:177]: Still, I doubt that we don't want to show secrets to the user

File: src/screens/ImportWalletScreen.tsx (hunk 5 above)
[the remark this answers is not in the document]
Comment (Kamill90) [reply src/screens/ImportWalletScreen.tsx:21]: Ok, I expected that more saga magic will be needed but this is fine for me

File: src/screens/ImportWalletScreen.tsx (hunk 8 above)
Comment (KedziaPawel) [reply src/screens/ImportWalletScreen.tsx:174]: yes, I answered you in the comment https://github.com/bitcoinvault/GoldWallet/pull/582#discussion_r560186037

**Review by marek-siemieniuk-morawski — APPROVED** [review]

**Review by KedziaPawel — APPROVED** [review]

**Pull request merged**